linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala

```diff
@@ -83,6 +89,35 @@ object ResponseClassifiers {
     }
   }
 
+  object GatewayErrorResponses {
+
+    object Failure {
+      def unapply(rsp: Response): Boolean = rsp.status match {
+        case Status.BadGateway | Status.ServiceUnavailable | Status.GatewayTimeout => true
+        case _ => false
+      }
+
+      // There are probably some (linkerd-generated) failures that aren't
+      // really retryable... For now just check if it's a failure.
+      object Retryable {
+        def unapply(rsp: Response): Boolean = Failure.unapply(rsp)
+      }
+    }
+  }
+
+  object RetryableGatewayErrorResult {
```

Comment:
Could you please add some comments here to clarify what this code does? Thanks.

Reply:
Added